portfolio/src/main/java/com/google/sps/servlets/CreateNewComment.java

```diff
@@ -31,17 +33,32 @@
  */
 @WebServlet("/create-new-comment")
 public class CreateNewComment extends HttpServlet {  
-  
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String username = request.getParameter("username"); 
-    String commentContent = request.getParameter("comment"); 
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html"); // set response type
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      response.getWriter().println("Hello " + userEmail + "! Now that you're logged in, feel free to leave a comment.");   
+    } else {
+      response.getWriter().println("Please log in to leave a comment.");   
+    }
+  }
 
-    Entity commentEntity = Comment.createNewCommentEntity(username, commentContent);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(); // create instance of DatastoreService class
-    datastore.put(commentEntity);
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Only logged-in users can add a comment
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String username = request.getParameter("username"); 
+      String commentContent = request.getParameter("comment"); 
+      response.setContentType("text/html"); 
+      Entity commentEntity = Comment.createNewCommentEntity(username, commentContent);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(); // create instance of DatastoreService class
+      datastore.put(commentEntity);
+    } 
```

Comment:
So what is the server response if the person isn't logged in?
You should set a response code that signifies unathorized access.
In the HTTP spec, thats's response code 401/403. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

Reply:
I have set a response code